home_news

evacalla opened the pull request:
- [X] Agrego configuración para iphone
- [X] Defino los file home_show, news_element_show y new_show  para mostrar el contenido de las noticias
- [X] Agrego servicio de newApi y tambien su dependencia
- [X] Agrego al readme la dependencia

No soy una mala persona por no aceptar los return, solo he tenido mala suerte - Patoruzito

evacalla committed:
home_news

Agrego configuración para iphone
Defino los file home_show, news_element_show y new_show  para mostrar el contenido de las noticias
Agrego servicio de newApi y tambien su dependencia
Agrego al readme la dependencia

matiasm15 commented on lib/screens/home/widgets/home_show.dart:
```diff
@@ -0,0 +1,97 @@
+        padding: const EdgeInsets.all(15.0),
+          child: Column (
+            mainAxisAlignment: MainAxisAlignment.center,
+            // ignore: prefer_const_literals_to_create_immutables
```
Por que lo ignoras? jajaja. El IDE te dice como arreglarlo.

matiasm15 commented on lib/screens/home/widgets/home_show.dart:
```diff
@@ -0,0 +1,97 @@
+import 'package:yisty_app/services/rest_client/api_exceptions.dart';
+import 'package:yisty_app/widgets/design/alert_page.dart';
+import 'package:yisty_app/widgets/design/empty_page.dart';
+import 'news_element_show.dart';
```
Usaria la forma recomendada por el IDE,
```
import 'package:yisty_app/screens/home/widgets/news_element_show.dart';
```

matiasm15 commented on lib/screens/home/widgets/home_show.dart:
```diff
@@ -0,0 +1,97 @@
+            return buildNoResults();
+          }
+
+          return SingleChildScrollView(
```
Para que un SingleChildScrollView? El BasicScaffold ya incluye uno.

evacalla replied:
Si lo vi, pero no me funcionaba a nivel de página entonces se lo agregue

matiasm15 commented on lib/screens/home/widgets/home_show.dart:
```diff
@@ -0,0 +1,97 @@
+  Widget buildList(BuildContext context) {
+    return FutureBuilder<List<Article>>(
+      // cambiar la alberto cuando este las rectricciones
+      future: newsApiService.getArticleResponseByEverything('alberto', 'ar', 30, 'es'),
```
Los futures no se deben generar dentro de un build. La forma correcta es generarlo dentro de un `initState` o en un `didChangeDependencies`. Fijate como lo hice en https://github.com/matiasm15/yisty-app/pull/6/files#diff-58e628551d8e1d56800611179e9351181bdc49e623a75979b809d08d21db2f37

evacalla replied:
A que comentario hago caso al de abajo o este? decidite eso por tener dos returns

matiasm15 commented on lib/screens/home/widgets/news_show.dart:
```diff
@@ -0,0 +1,62 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```
No podes usar BasicScaffold? Para mantener un mismo Scaffold en todas las pantallas.

evacalla replied:
Cierto

matiasm15 commented on lib/services/new_api_service.dart:
```diff
@@ -0,0 +1,29 @@
+
+class NewsApiService {
+
+  final String _apiKey = '942b019bd36e4ab28f68930032848aa7';
```
Habria que mandarlo a una variable de entorno.

evacalla replied:
el clean matias

matiasm15 commented on lib/screens/home/widgets/home_show.dart:
```diff
@@ -0,0 +1,97 @@
+  Widget buildList(BuildContext context) {
+    return FutureBuilder<List<Article>>(
+      // cambiar la alberto cuando este las rectricciones
+      future: newsApiService.getArticleResponseByEverything('alberto', 'ar', 30, 'es'),
```
Podrias usar `InheritedProvider.of(context).uiStore.user.profile.name`.

evacalla replied:
Ready

evacalla committed:
home_news | cambios por pr

matiasm15 commented on lib/screens/home/widgets/home_show.dart:
```diff
@@ -0,0 +1,103 @@
+class _HomeShowState extends State<HomeShow> {
+
+  Future<List<Article>> _future;
+  String preference;
```
Esto puede ser una variable local en didChangeDependencies(), no necesita ser de la clase.

evacalla replied:
Ready?

matiasm15 commented on lib/services/new_api_service.dart:
```diff
@@ -0,0 +1,30 @@
+      return response.articles;
+  }
+
+  Future<List<Article>> getArticleResponseByEverything (String search, String country, int pageSize, String language) async {
```
Para que recibe tantos parametros si solo usa `search`?

evacalla replied:
Podria definir los parametros como cte dentro del servicio, estos parametros se utiliza para especificar, el idioma, region, sarasa

evacalla committed:
home_news | cambio por pr

Defino preference dentro del metodo
Retiro parametros opcionales de busqueda de newapi
